File: ai-service/app/schemas/patient.py
from datetime import date
from typing import Optional

from pydantic import BaseModel, EmailStr, Field


class PatientCreate(BaseModel):
    first_name: str = Field(min_length=1, max_length=50)
    last_name: str = Field(min_length=1, max_length=50)

    date_of_birth: date

    sex: str

    phone_number: str

    email: Optional[EmailStr] = None

    address_line_1: str
    address_line_2: Optional[str] = None

    city: str = Field(min_length=1, max_length=100)
    state: str = Field(min_length=2, max_length=2)
    zip_code: str

    insurance_provider: Optional[str] = None
    insurance_member_id: Optional[str] = None

    preferred_language: str = "English"

    emergency_contact_name: Optional[str] = None
    emergency_contact_phone: Optional[str] = None

File: frontend/react-app/src/types/patient.ts
export interface Patient {
  patient_id: string;
  first_name: string;
  last_name: string;
  date_of_birth: string; // YYYY-MM-DD
  sex: 'Male' | 'Female' | 'Other' | 'Decline to Answer';
  phone_number: string;
  email?: string | null;
  address_line_1: string;
  address_line_2?: string | null;
  city: string;
  state: string;
  zip_code: string;
  insurance_provider?: string | null;
  insurance_member_id?: string | null;
  preferred_language: string;
  emergency_contact_name?: string | null;
  emergency_contact_phone?: string | null;
  created_at: string;
  updated_at: string;
  deleted_at?: string | null;
}

export type PatientCreateRequest = Omit<
  Patient,
  'patient_id' | 'created_at' | 'updated_at' | 'deleted_at'
>;

export type PatientUpdateRequest = Partial<PatientCreateRequest>;

export interface ApiResponse<T> {
  data: T | null;
  error: string | Record<string, string[]> | null;
}
